cpp/src/arrow/compute/kernels/vector_run_end_encode.cc

```diff
@@ -508,6 +509,10 @@ static ArrayKernelExec GenerateREEKernelExec(Type::type type_id) {
       return Functor::template Exec<UInt64Type>;
     case Type::INTERVAL_MONTH_DAY_NANO:
       return Functor::template Exec<MonthDayNanoIntervalType>;
+    case Type::DECIMAL32:
+      return Functor::template Exec<Decimal32Type>;
+    case Type::DECIMAL64:
+      return Functor::template Exec<Decimal64Type>;
```

Comment:
I'm curious, do we have to instantiate `Exec` for these additional types? Or can we reuse `Exec<UInt32Type>` and `Exec<UInt64Type>` respectively? That would reduce the amount of code generated for these kernels.

Reply:
Thanks for the ping @pitrou. That's something I hadn't thought of. The idea makes sense, let me test.

Reply:
Added in 3c5d7958af8e89a1aa9ce833e1387d82920e7bda, tests still pass locally so that's good.